lib/logstash/filters/elasticsearch.rb

```diff
@@ -126,13 +126,7 @@ class LogStash::Filters::Elasticsearch < LogStash::Filters::Base
   config :tag_on_failure, :validate => :array, :default => ["_elasticsearch_lookup_failure"]
 
   def register
-    options = {
-      :ssl => @ssl,
-      :hosts => @hosts,
-      :ca_file => @ca_file,
-      :logger => @logger
-    }
-    @client = LogStash::Filters::ElasticsearchClient.new(@user, @password, options)
+    @clients_pool = {}
```

Comment:
This isn't used. We should, however, use this as opposed to `Thread.current[:filter_elasticsearch_client]` since we'll need to close all the clients on pipeline shutdown.

I recommend making this a `java.util.concurrent.ConcurrentHashMap` indexed by `Thread.current`, and using that to handle the close behavior.

Reply:
Funny, It was my first draft yesterday. I've figured since we already have `Thread.current`'s hash it would have been enough.